### PR TITLE
10.3.0 consolidated saved search

### PR DESCRIPTION
**Pull request by weslyvo81**
This pull request is about cleaning up and improving saved search based on design doc.

### CODE REVIEW

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java [hunk 1]
```diff
@@ -94,18 +88,7 @@ public  boolean isValueSelected(String code) {
      */
     public  boolean isApply(String code) {
```
Comment (jasperRob) [inline kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java:91]: This should be:
```java
public boolean isApply(String code) {
    return Question.QUE_SAVED_SEARCH_APPLY.equalsIgnoreCase(code);
}
```

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java [hunk 2]
```diff
@@ -117,8 +100,8 @@ public  boolean isValidTable(String code) {
      */
     public boolean isColumnSelected(String code,String attCode) {
         boolean result = false;
```
Comment (jasperRob) [inline kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java]: This should be:
```java
public boolean isColumnSelected(String code,String attCode) {
    if (Question.QUE_FILTER_COLUMN.equalsIgnoreCase(code)) {
        return true;
    }
    if (Attribute.LNK_FILTER_COLUMN.equalsIgnoreCase(attCode)) {
        return true;
    }
    return false;
```

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java [hunk 3]
```diff
@@ -131,8 +114,8 @@ public boolean isColumnSelected(String code,String attCode) {
      */
     public boolean isOptionSelected(String code, String attCode) {
         boolean result = false;
```
Comment (jasperRob) [inline kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java]: This needs a clean up.

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java [hunk 4]
```diff
@@ -427,9 +245,7 @@ public String selectFilerColumn(String value) {
      */
     public boolean isBtnSearchSave(String code) {
```
Comment (jasperRob) [inline kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java:248]: This needs a clean up.

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java [hunk 5]
```diff
@@ -440,9 +256,7 @@ public boolean isBtnSearchSave(String code) {
      */
     public boolean isBtnSearchDelete(String code) {
```
Comment (jasperRob) [inline kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java:257]: This needs a clean up.

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java [hunk 6]
```diff
@@ -453,9 +267,7 @@ public boolean isBtnSearchDelete(String code) {
      */
     public boolean isDetailDelete(String code) {
```
Comment (jasperRob) [inline kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java:266]: This needs a clean up.

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java [hunk 7]
```diff
@@ -572,7 +382,7 @@ public String convertLnkArrayToString(List<String> params) {
      * @param code Base entity
      */
     public void deleteSearch(String code) {
-        databaseUtils.deleteBaseEntityAndAttribute(userToken.getProductCode(), code);
+        databaseUtils.deleteBaseEntityAndAttribute(user.getProductCode(), code);
```
Comment (jasperRob) [inline kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java:381]: Don't delete the BaseEntity, just update the EEntityStatus to DELETED

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java [hunk 8]
```diff
@@ -769,9 +562,7 @@ public void putAnswerstoCache(String attCode,String value) {
      * @return Question code
      */
     public String getQuestionCode(QEventMessage msg) {
```
Comment (jasperRob) [inline kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java:560]: This method seems redundant, we should not be returning an empty string.

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java [hunk 9]
```diff
@@ -781,9 +572,7 @@ public String getQuestionCode(QEventMessage msg) {
      * @return Question code
      */
     public String getQuestionCode(QDataAnswerMessage msg) {
```
Comment (jasperRob) [inline kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java]: This method seems redundant, we should not be returning an empty string.

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java [hunk 10]
```diff
@@ -793,9 +582,7 @@ public String getQuestionCode(QDataAnswerMessage msg) {
      * @return Attribute code
      */
     public String getAttributeCode(QDataAnswerMessage msg) {
```
Comment (jasperRob) [inline kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java]: We should not be returning an empty string. Also, consider the case that more than one item is present.

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java [hunk 11]
```diff
@@ -805,9 +592,7 @@ public String getAttributeCode(QDataAnswerMessage msg) {
      * @return target code
      */
     public String getTargetCode(QDataAnswerMessage msg) {
```
Comment (jasperRob) [inline kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java]: We should not be returning an empty string. Also, consider the case that more than one item is present.

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java [hunk 12]
```diff
@@ -817,9 +602,7 @@ public String getTargetCode(QDataAnswerMessage msg) {
      * @return value
      */
     public String getValue(QDataAnswerMessage msg) {
```
Comment (jasperRob) [inline kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java]: We should not be returning an empty string. Also, consider the case that more than one item is present.

File: kogitoq/kogito-common/src/main/java/life/genny/kogito/common/service/FilterService.java [hunk 13]
```diff
@@ -192,24 +51,21 @@ public void handleQuickSearchDropdown(String value,boolean coded,String targetCo
...
         // searching by text or search by code
         Filter filter = null;
         String newValue = value.replaceFirst("!","");
-        if(coded) {
-            filter = new Filter(Attribute.PRI_CODE, Operator.EQUALS, value);
-        }else {
-            filter = new Filter(Attribute.PRI_NAME, Operator.LIKE, "%" + newValue + "%");
-        }
+
+        if(coded) filter = new Filter(Attribute.PRI_CODE, Operator.EQUALS, value);
```
Comment (jasperRob) [inline kogitoq/kogito-common/src/main/java/life/genny/kogito/common/service/FilterService.java]: Don't use one line if statements, it goes against the google java style guide.

**Review by jasperRob — CHANGES_REQUESTED** [review]
Please address these changes. Most of then are cleaning up some methods.

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java (hunk 1 above)
Comment (weslyvo81) [reply kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java:91]: solved

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java (hunk 9 above)
Comment (weslyvo81) [reply kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java]: it is used in other methods

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java (hunk 11 above)
Comment (weslyvo81) [reply kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java]: just get the first item

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java (hunk 10 above)
Comment (weslyvo81) [reply kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java]: just get the first item.

File: kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java (hunk 12 above)
Comment (weslyvo81) [reply kogitoq/gadaq/src/main/java/life/genny/gadaq/search/FilterGroupService.java]: just get the first item.

**Review by jasperRob — APPROVED** [review]
This will need a little more clean up in future, but good for now.

**Review by varunshastry — APPROVED** [review]

**Pull request merged**